lib/bap_llvm/bap_llvm_ogre_loader.ml

```diff
@@ -82,28 +79,38 @@ module Make(M : Monad.S) = struct
     S.symbols  >>= fun () ->
     S.code_regions
 
-  let image = Dispatcher.of_filetype >>= provide
+  let image = of_filetype >>= provide
 
 end
 
-module Loader = struct
+module Loader(P : Parameters) = struct
 
-  module Ogre_loader = Make(Monad.Ident)
+  module Ogre_loader = Ogre_loader(P)
   open Ogre_loader
 
   exception Llvm_loader_fail of int
 
   let _ = Callback.register_exception
       "Llvm_loader_fail" (Llvm_loader_fail 0)
 
+  let pdb_file filename =
+    let file =
+      if Sys.is_directory P.pdb_path then
+        let pdb_file = sprintf "%s.pdb"
+            Filename.(remove_extension @@ basename filename) in
+        Filename.concat P.pdb_path pdb_file
+      else P.pdb_path in
+    if Sys.file_exists file then file
```

Comment:
`Sys.is_directory p` raises an exception if `p` doesn't exist. Therefore this code is unreachable. You should first check if it exists, then if it is, check if it is a directory or not.